internal/endorser/endorser.go

```diff
@@ -37,7 +38,9 @@ import (
 // valid. Each provenanceURI must either specify a local file (using the `file` scheme), or a
 // remote file (using the `http/https` scheme).
 func GenerateEndorsement(binaryDigest string, validityDuration amber.ClaimValidity, provenanceURIs []string) (*intoto.Statement, error) {
-	verifiedProvenances, err := loadAndVerifyProvenances(provenanceURIs, binaryDigest)
+	expected := verifier.ProvenanceIR{BinarySHA256Digest: []string{binaryDigest}}
```

Comment:
How about instead passing `expected verifier.ProvenanceIR` as input to `GenerateEndorsement`? So that the caller of `GenerateEndorsement` is responsible for creating an instance of `verifier.ProvenanceIR`.